packages/devextreme/js/__internal/ui/m_tile_view.ts

```diff
@@ -203,69 +212,90 @@ class TileView extends CollectionWidget<TileViewProperties> {
   }
 
   _renderGeometry(): void {
-    const { direction } = this.option();
-    // @ts-expect-error ts-error
+    const { direction = 'horizontal' } = this.option();
+
     this._config = CONFIGS[direction];
 
-    const items = this.option('items') || [];
-    const config = this._config;
-    const itemMargin = this.option('itemMargin');
-    const maxItemCrossRatio = Math.max.apply(Math, map(items || [], (item) => Math.round(item[config.itemCrossRatio] || 1)));
+    const { items = [], itemMargin = 0 } = this.option();
+    const config: OrientationConfig<Orientation> = this._config;
 
+    const maxItemCrossRatio = items.reduce(
+      (max, item) => {
+        const ratio = Math.round(item[config.itemCrossRatio] ?? 1);
+        return Math.max(max, ratio);
+      },
+      1,
+    );
+
+    // eslint-disable-next-line @typescript-eslint/init-declarations
     let crossDimensionValue;
-    // @ts-expect-error
+    // @ts-expect-error ts-error
     if (hasWindow) {
       crossDimensionValue = (config.crossDimension === 'width' ? getWidth : getHeight)(this.$element());
     } else {
-      // @ts-expect-error ts-error
-      // eslint-disable-next-line radix
-      crossDimensionValue = parseInt(this.$element().get(0).style[config.crossDimension]);
+      crossDimensionValue = parseInt(this.$element()[0].style[config.crossDimension], 10);
     }
-    // @ts-expect-error ts-error
-    this._cellsPerDimension = Math.floor(crossDimensionValue / (this.option(config.baseItemCrossDimension) + itemMargin));
+
+    const options = this.option();
+    const baseItemCrossDimension = options[config.baseItemCrossDimension] ?? 0;
+
+    this._cellsPerDimension = Math.floor(
+      crossDimensionValue / (baseItemCrossDimension + itemMargin),
+    );
     this._cellsPerDimension = Math.max(this._cellsPerDimension, maxItemCrossRatio);
     this._cells = [];
-    // @ts-expect-error ts-error
+
     this._cells.push(new Array(this._cellsPerDimension));
 
     this._arrangeItems(items);
     this._renderContentSize(config, itemMargin);
   }
 
-  _renderContentSize(config, itemMargin): void {
+  _renderContentSize(
+    config: OrientationConfig<Orientation>,
+    itemMargin: number | undefined = 0,
```

Comment:
in this particular case may be simplified
```suggestion
    itemMargin = 0,
```